src/pages/Landing/index.tsx

```diff
@@ -300,35 +301,28 @@ const Link = styled(NativeLink)`
 
 export default function Landing() {
   const isDarkMode = useIsDarkMode()
-
   const cardsRef = useRef<HTMLDivElement>(null)
-
-  const [showContent, setShowContent] = useState(false)
   const selectedWallet = useAppSelector((state) => state.user.selectedWallet)
-  const navigate = useNavigate()
-  const location = useLocation()
-  const queryParams = parse(location.search, {
-    ignoreQueryPrefix: true,
-  })
+  const shouldDisableNFTRoutes = useDisableNFTRoutes()
+  const cards = useMemo(
+    () => MAIN_CARDS.filter((card) => !(shouldDisableNFTRoutes && card.to.startsWith('/nft'))),
+    [shouldDisableNFTRoutes]
+  )
 
   const [accountDrawerOpen] = useAccountDrawer()
+  const navigate = useNavigate()
   useEffect(() => {
-    if ((queryParams.intro || !selectedWallet) && !accountDrawerOpen) {
-      setShowContent(true)
-    } else {
-      setShowContent(false)
+    if (accountDrawerOpen) {
       setTimeout(() => {
         navigate('/swap')
-      }, TRANSITION_DURATIONS.medium)
+      }, TRANSITION_DURATIONS.fast)
     }
-  }, [navigate, selectedWallet, queryParams.intro, accountDrawerOpen])
+  }, [accountDrawerOpen, navigate])
 
-  const shouldDisableNFTRoutes = useDisableNFTRoutes()
-
-  const cards = useMemo(
-    () => MAIN_CARDS.filter((card) => !(shouldDisableNFTRoutes && card.to.startsWith('/nft'))),
-    [shouldDisableNFTRoutes]
-  )
+  const queryParams = parse(useLocation().search, { ignoreQueryPrefix: true })
```

Comment:
nit: I find inline hook usage a little harder to parse at first glance and prefer the prior 2-line setup